### PR TITLE
oauth filter: set token cookies regardless of forward_bearer_token option and ability to disable bearertoken and refreshtoken cookie

### PR DESCRIPTION
**Pull request by denniskniep**
**Commit Message:**
oauth filter: set token cookies regardless of forward_bearer_token option + ability to disable    refreshtoken and bearertoken cookie

**Additional Description:** 
Unconditionally set the BearerToken, IdToken, and RefreshToken cookies in the response. The documentation of forward_bearer_token states "Forward the OAuth token as a XXX to upstream web service." It's confusing for this behavior to affect response cookies as well.

This change alone would set the raw bearer token in the client browser which is undesirable by some.

Therefore introduced further properties to disable single cookies, if necessary:
* `disable_access_token_set_cookie`
* `disable_refresh_token_set_cookie`

Like it was done here: https://github.com/envoyproxy/envoy/issues/33825

Risk Level: Low
Testing: Included
Docs Changes: N/A
Release Notes: Included
Platform Specific Features: N/A
Fixes: https://github.com/envoyproxy/envoy/issues/32566

### CODE REVIEW

File: api/envoy/extensions/filters/http/oauth2/v3/oauth.proto [hunk 1]
```diff
@@ -166,6 +166,18 @@ message OAuth2Config {
   // will still process incoming ID Tokens as part of the HMAC if they are there. This is to ensure compatibility while switching this setting on. Future
   // sessions would not set the IdToken cookie header.
   bool disable_id_token_set_cookie = 17;
+
+  // If set to true, Envoy will not set a cookie for Access Token even if one is received from the Identity Provider.
```
Comment (markdroth) [inline api/envoy/extensions/filters/http/oauth2/v3/oauth.proto:170]: s/Envoy/the client/

Envoy is not the only xDS client.

File: api/envoy/extensions/filters/http/oauth2/v3/oauth.proto [hunk 2]
```diff
@@ -166,6 +166,18 @@ message OAuth2Config {
...
+  // Enabling this option will only disable setting the cookie response header, the filter
+  // will still process incoming Access Tokens as part of the HMAC if they are there. This is to ensure compatibility while switching this setting on. Future
+  // sessions would not set the Access Token cookie header.
+  bool disable_access_token_set_cookie = 18;
+
+  // If set to true, Envoy will not set a cookie for Refresh Token even if one is received from the Identity Provider.
```
Comment (markdroth) [inline api/envoy/extensions/filters/http/oauth2/v3/oauth.proto:176]: s/Envoy/the client/

**Comment by denniskniep** [comment]
@markdroth I addressed your review comments

**Comment by markdroth** [comment]
In the future, please don't force-push to the branch after the review has started, since that makes it hard for the reviewer to see what changed since their last review pass.

**Comment by denniskniep** [comment]
cc: [derekargueta](https://github.com/derekargueta)

**Review by mattklein123 — DISMISSED** [review]

**Comment by denniskniep** [comment]
@mattklein123 thanks for the review. Can we already merge it?

**Comment by mattklein123** [comment]
I'm not sure why CI is failing. Is main fully merged? cc @phlax

**Comment by denniskniep** [comment]
Hi @mattklein123 ,

merged `main` again and removed that `+` char from the pr title. I think removing the `+` char did the trick...
Can we merge this pr?


Probably [this](https://github.com/envoyproxy/envoy/blob/4d99ab1627a9b847b6cd5fba3e4a816b1f5b883c/.github/workflows/_start.yml#L90) line can not handle it ?
> The template is not valid. envoyproxy/envoy/.github/workflows/_start.yml (Line: 90, Col: 17): Error reading JToken from JsonReader. Path '', line 0, position

**Comment by phlax** [comment]
hi @denniskniep - i think it was something slightly different

i changed `Bearer to` in the PR description to `XXX` - its a new failure mode triggered by a false positive in githubs secret scanning

**Comment by mattklein123** [comment]
Sorry needs a main merge then we are good to go.

/wait

**Comment by denniskniep** [comment]
@mattklein123  merged

**Comment by mattklein123** [comment]
Sorry needs merge again. :(

/wait

**Review by phlax — APPROVED** [review]
approving on basis of @mattklein123 's prior approval

lgtm, thanks @denniskniep

**Pull request merged**